### PR TITLE
Intermediate release 0.12.2

### PR DESCRIPTION
**Pull request by maxfischer2781**
This release publishes accumulated changes and fixes. Notable changes include:

* #87
* #91
* #92

Auxiliary changes:

* [x] Current lint standards
* [x] Change log regenerated
* [x] Version bump

If merged, a new release will be published to PyPI.

### CODE REVIEW

**Comment by maxfischer2781** [comment]
Heads up that there are some spurious test failures – see #95 for tracking. They don't seem related to the code changes, nor should they be relevant for production.

I would prefer to make this release *before* looking into the failures so that current bugfixes are available promptly. Fixing the failures may involve larger changes that are more suitable for a "major" release instead.

File: src/cobald/daemon/config/mapping.py
Comment (giffels) [inline src/cobald/daemon/config/mapping.py:99]: Should it not link to the "`key_error`" above?

File: src/cobald/daemon/core/logger.py
Comment (giffels) [inline src/cobald/daemon/core/logger.py:24]: Should it not link to the "`attribute_error`" above?

**Review by giffels — CHANGES_REQUESTED** [review]
Thanks a lot for your work. It looks fine to me. However, could you comment on my inline comments, please? This not a show stopper, however I would like have this discussed before the approval.

File: src/cobald/daemon/config/mapping.py
Comment (maxfischer2781) [reply src/cobald/daemon/config/mapping.py:99]: The ``KeyError`` is just an implementation detail here of modules being stored in a dict. I don't think it would add any relevant information, especially since the missing key is reported in the ``ImprotError`` as well.

File: src/cobald/daemon/core/logger.py
Comment (maxfischer2781) [reply src/cobald/daemon/core/logger.py:24]: Since this is called to process "user" input, I don't think the entire trace is relevant. The purpose is to preserve for the user that they used an invalid log level; knowing that we tried fetching it from ``logging`` doesn't seem to add anything of use for them.

Comment (giffels) [reply src/cobald/daemon/core/logger.py:24]: Thanks a lot. Agreed

**Review by giffels — APPROVED** [review]
Looks good to go. Thanks for the contribution.

**Review by eileen-kuehn — APPROVED** [review]
Looks good from my point of view. Go for it 🥇 and push for pypi.

**Pull request merged**
